frappe/model/meta.py

```diff
@@ -452,14 +448,56 @@ def init_field_caches(self):
 			self._table_fields = self.get("fields", {"fieldtype": ["in", table_fields]})
 
 	def sort_fields(self):
-		"""Sort custom fields on the basis of insert_after"""
+		"""
+		Sort fields on the basis of following rules (priority descending):
+		- `field_order` property setter
+		- `insert_after` computed based on default order for standard fields
+		- `insert_after` property for custom fields
+		"""
+
+		if field_order := getattr(self, "field_order", []):
+			field_order = [fieldname for fieldname in json.loads(field_order) if fieldname in self._fields]
+
+			# all fields match, best case scenario
+			if len(field_order) == len(self.fields):
+				self._update_fields_based_on_order(field_order)
+				return
+
+			# if the first few standard fields are not in the field order, prepare to prepend them
+			if self.fields[0].fieldname not in field_order:
+				fields_to_prepend = []
+				standard_field_found = False
+
+				for fieldname, field in self._fields.items():
+					if getattr(field, "is_custom_field", False):
+						# all custom fields from here on
+						break
+
```

Comment:
The case where custom field is added to the top (in case it isn't in field order) is handled in unchanged line 507: https://github.com/frappe/frappe/pull/19822/files#diff-f23ec45599e5f8df9263a49b71658cd63d3442008ffa46e3f8aeb1b97d571556R507

This algorithm isn't meant to perfect, just _good enough_. It is possible that the custom field meant to be on the top is in the field order and the new standard fields added to the doctype will go above that. I don't think this algorithm can be perfected in the current design, which is why I had added an optional implementation of notifying users when their field order might need updates.